extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosImporter.java

```diff
@@ -297,32 +294,32 @@ long importPhotoBatch(
 
   private long processMediaResult(
       NewMediaItemResult mediaItem,
-      String idempotentId,
+      ImportableItem item,
       IdempotentImportExecutor executor,
-      String title,
       long bytes)
       throws Exception {
     Status status = mediaItem.getStatus();
     if (status.getCode() == Code.OK_VALUE) {
-      executor.executeAndSwallowIOExceptions(
-          idempotentId, title, () -> new PhotoResult(mediaItem.getMediaItem().getId(), bytes));
+      PhotoResult photoResult = new PhotoResult(mediaItem.getMediaItem().getId(), bytes);
+      executor.importAndSwallowIOExceptions(
+          item, inp -> ItemImportResult.success(photoResult, bytes));
       return bytes;
     } else {
-      executor.executeAndSwallowIOExceptions(
-          idempotentId,
-          title,
-          () -> {
-            throw new IOException(
-                String.format(
-                    "Media item could not be created. Code: %d Message: %s",
-                    status.getCode(), status.getMessage()));
-          });
+      executor.importAndSwallowIOExceptions(
+          item,
+          inp ->
```

Comment:
nit: I'm not sure what `inp` is an acronym for. Could you give a more descriptive name to `inp` variable - both here and elsewhere like line 305? eg: maybe `itemToImport` or something, assuming I'm understanding correctly.